### PR TITLE
Yash arrow key control

### PR DESCRIPTION
**Pull request by anedergaard**
Support for home, end, delete, left/right arrow and ctrl+left/right keys

### CODE REVIEW

**Review by thomas-mp — COMMENTED** [review]
Nice Anders, you did it :p

I will leave it to the other reviewers to approve it ;)

File: include/Yash.h [hunk 1]
```diff
@@ -299,10 +449,13 @@ class Yash {
     static constexpr const char* s_clearLine = "\033[2K\033[100D";
     static constexpr const char* s_clearScreen = "\033[2J\x1B[H";
     static constexpr const char* s_clearCharacter = "\033[1D \033[1D";
+    static constexpr const char *s_moveCursorForward = "\033[1C";
+    static constexpr const char *s_moveCursorBackward = "\033[1D";
     static constexpr const char* s_commandDelimiter = " ";
     std::map<std::string, Function> m_functions;
     std::vector<std::string> m_commands;
     std::vector<std::string>::const_iterator m_commandsIndex;
+    size_t pos{0};
```
Comment (jeppefrandsen) [inline include/Yash.h]: Maybe m_position would be better? Think we also miss clang format for the missing space

File: include/Yash.h [hunk 2]
```diff
@@ -206,6 +334,28 @@ class Yash {
...
+        Down,
+        Right,
+        Left,
+        Home,
+        Delete,
+        End,
+        CtrlRight,
+        CtrlLeft
+    };
+    const std::map<std::string, CtrlSeq> SupportedCtrlSeq = {
```
Comment (jeppefrandsen) [inline include/Yash.h]: Think these should be named with `m_` also and probably be at the bottom since they are variables

**Review by jeppefrandsen — COMMENTED** [review]
Awesome @anedergaard 🍻. A couple of minor comments

File: include/Yash.h [hunk 3]
```diff
@@ -130,21 +130,41 @@ class Yash {
...
         case Del:
         case Backspace:
             if (m_command.size()) {
-                print(s_clearCharacter);
-                m_command.erase(m_command.size() - 1);
+                if (pos == m_command.length()) {
+                    print(s_clearCharacter);
+                    m_command.erase(m_command.length() - 1);
+                    pos = m_command.length();
+                } else if (pos != 0) {
```
Comment (jeppefrandsen) [inline include/Yash.h]: Just `(pos)` according to WebKit (multiple places)

Comment (anedergaard) [reply include/Yash.h]: should be fixed

File: include/Yash.h (hunk 2 above)
Comment (anedergaard) [reply include/Yash.h]: should be fixed

File: include/Yash.h (hunk 1 above)
Comment (anedergaard) [reply include/Yash.h]: should be fixed

**Review by jeppefrandsen — APPROVED** [review]

**Pull request merged**